Sources/TuistLoader/Utils/ManifestFilesLocator.swift

```diff
@@ -108,38 +121,30 @@ public final class ManifestFilesLocator: ManifestFilesLocating {
             ]
             .filter { FileHandler.shared.exists($0.path) }
         } else {
-            guard let rootPath = rootDirectoryLocator.locate(from: locatingPath) else {
-                let projectsPaths = FileHandler.shared.glob(locatingPath, glob: "**/\(Manifest.project.fileName(locatingPath))")
-                    .map {
-                        ProjectManifest(
-                            manifest: Manifest.project,
-                            path: $0
-                        )
-                    }
-                let workspacesPaths = FileHandler.shared.glob(locatingPath, glob: "**/\(Manifest.workspace.fileName(locatingPath))")
-                    .map {
-                        ProjectManifest(
-                            manifest: Manifest.workspace,
-                            path: $0
-                        )
-                    }
-                return projectsPaths + workspacesPaths
-            }
-
-            let projectsPaths = FileHandler.shared.glob(rootPath, glob: "**/\(Manifest.project.fileName(rootPath))")
+            let path = rootDirectoryLocator.locate(from: locatingPath) ?? locatingPath
+
+            let excludedPaths = excluding
+                .flatMap {
+                    FileHandler.shared.glob(path, glob: $0)
+                }
+
+            let projectsPaths = Set(FileHandler.shared.glob(path, glob: "**/\(Manifest.project.fileName(path))"))
```

Comment:
Nitpick: If you subtract to the union of `projectPaths` and `workspacePaths` it'll be slightly more performant because it'll be one less operation:
`(projectPaths + workspacePaths).subtracting(excludedPaths)`